src/main/java/org/apache/mesos/api/JettyApiServer.java

```diff
@@ -0,0 +1,55 @@
+package org.apache.mesos.api;
+
+import org.apache.mesos.scheduler.plan.api.PlanResource;
+import org.apache.mesos.scheduler.recovery.api.RecoveryResource;
+import org.apache.mesos.state.api.StateResource;
+import org.eclipse.jetty.server.Server;
+import org.glassfish.jersey.jetty.JettyHttpContainerFactory;
+import org.glassfish.jersey.server.ResourceConfig;
+
+import javax.ws.rs.core.UriBuilder;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ * A JettyApiServer takes a list of POJO JAX-RS Resources and serves them on the indicated port of 0.0.0.0.
+ */
+public class JettyApiServer {
```

Comment:
Why create this wrapper ? Bootstrapping and configuring jetty is fairly trivial. By creating this wrapper layer, we are taking away all the native config options our developers can use.